### ISSUE TITLE
Memory allocation error in Page.getText()

### ISSUE THREAD

**Issue by nxhoehing** [pymupdf/PyMuPDF#710]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
When extracting Text from 0th page of the pdf in the attachment, pyymupdf gets stuck in a loop.
## To Reproduce (mandatory)
Explain the steps to reproduce the behavior, For example, include a minimal code snippet, example files, etc.
path = '/content/drive/My Drive/PDF/NYSE_DAI_2018.pdf'
pdf = fitz.open(path)
for page in pdf.pages(0,1):
  print(page.getText()) 
## Expected behavior (optional)
Describe what you expected to happen (if not obvious).
It would be preferable to get an error so that I can skip the page
## Your configuration (mandatory)
 - Operating system, potentially version and bitness
 - Python version, bitness
 - PyMuPDF version, installation method (**wheel** or **generated** from source).

3.6.9 (default, Jul 17 2020, 12:50:27) 
[GCC 8.4.0] 
 linux 
 
PyMuPDF 1.18.2: Python bindings for the MuPDF 1.18.0 library.
Version date: 2020-10-23 09:17:55.
Built for Python 3.6 on linux (64-bit).
For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

## Additional context (optional)
[NYSE_DAI_2019 Kopie.pdf](https://github.com/pymupdf/PyMuPDF/files/5457306/NYSE_DAI_2019.Kopie.pdf)

**Comment by JorjMcKie** [comment]
The problem has nothing to do with method `Document.pages()`.
It occurs in method `page.getText()` and it also occurs with MuPDF's batch program ``mutool draw -o page.txt NYSE_DAI_2019 Kopie.pdf 1``: after a considerable time that program crashes with a message on failure to allocate more memory.
Just tried cleaning the page - but the problem persists.
I suggest you submit it to MuPDF bug system directly: https://bugs.ghostscript.com/enter_bug.cgi. The problem here is that the file is too big to be accepted 🤔

**Comment by JorjMcKie** [comment]
The problem is also confined to page 0: text extraction from page 1 on works fine:
```python
>>> doc=fitz.open("NYSE_DAI_2019.Kopie.pdf")
>>> for page in doc.pages(1):
	print("text amount on page %i: %i" % (page.number, len(page.getText())))

	
text amount on page 1: 4123
text amount on page 2: 4433
text amount on page 3: 5302
text amount on page 4: 4709
text amount on page 5: 5726
text amount on page 6: 4318
text amount on page 7: 4723
text amount on page 8: 4695
text amount on page 9: 3575
text amount on page 10: 5311
text amount on page 11: 2744
text amount on page 12: 4924
text amount on page 13: 4126
text amount on page 14: 3022
text amount on page 15: 4497
text amount on page 16: 4526
text amount on page 17: 3225
text amount on page 18: 4473
text amount on page 19: 5561
text amount on page 20: 3234
text amount on page 21: 3722
text amount on page 22: 3046
text amount on page 23: 3388
text amount on page 24: 3769
text amount on page 25: 2224
text amount on page 26: 3275
text amount on page 27: 3030
text amount on page 28: 5542
text amount on page 29: 3818
text amount on page 30: 4137
text amount on page 31: 4936
text amount on page 32: 5103
text amount on page 33: 2265
text amount on page 34: 4463
text amount on page 35: 5583
text amount on page 36: 6801
text amount on page 37: 6960
text amount on page 38: 7399
text amount on page 39: 7263
text amount on page 40: 7089
text amount on page 41: 7166
text amount on page 42: 5540
text amount on page 43: 6079
text amount on page 44: 5261
text amount on page 45: 3575
text amount on page 46: 5272
text amount on page 47: 4999
text amount on page 48: 4999
text amount on page 49: 4358
text amount on page 50: 4998
text amount on page 51: 2602
text amount on page 52: 3319
text amount on page 53: 3926
text amount on page 54: 3983
text amount on page 55: 4680
text amount on page 56: 3356
text amount on page 57: 3448
text amount on page 58: 3997
text amount on page 59: 3754
text amount on page 60: 2735
text amount on page 61: 2426
text amount on page 62: 1457
text amount on page 63: 2934
text amount on page 64: 2662
text amount on page 65: 1433
text amount on page 66: 1577
text amount on page 67: 2823
text amount on page 68: 2481
text amount on page 69: 5676
text amount on page 70: 4168
text amount on page 71: 6222
text amount on page 72: 2475
text amount on page 73: 5779
text amount on page 74: 7045
text amount on page 75: 7499
text amount on page 76: 3939
text amount on page 77: 6816
text amount on page 78: 6404
text amount on page 79: 6318
text amount on page 80: 6397
text amount on page 81: 6532
text amount on page 82: 6549
text amount on page 83: 5933
text amount on page 84: 7014
text amount on page 85: 6084
text amount on page 86: 6742
text amount on page 87: 5650
text amount on page 88: 6117
text amount on page 89: 5260
text amount on page 90: 3215
text amount on page 91: 3262
text amount on page 92: 4315
text amount on page 93: 5541
text amount on page 94: 4662
text amount on page 95: 5524
text amount on page 96: 5679
text amount on page 97: 71
>>> 
```

**Comment by JorjMcKie** [comment]
Several PDF viewers have issues with that page:
Adobe Acrobat: ok
MuPDF viewer: loop
SumatraPDF: loop
Foxit Reader 9.7: loop
Microsft Edge browser: loop (empty page)
Firefox browser: ok
Nitro Reader 5: ok
PDF Xchange Reader: ok

**Comment by nxhoehing** [comment]
Hi, thanks a lot for your effort! I have just submitted the bug report and will get back to you, when I get a reply there.

**Comment by JorjMcKie** [comment]
> I have just submitted the bug report and will get back to you, when I get a reply there.

@nxhoehing - any rection from MuPDF yet?

**Comment by nxhoehing** [comment]
I got a reaction, but it does not seem like they will fix it: 
"The hanging and crashing has nothing to do with text extraction. It happens whenever we try to draw page 1 as well. Even the 'trace' device takes forever.

mutool clean -c and -s finish processing, so I suspect this may be down to a problem with interpreting the file. Looking at stack traces during execution, it seems to be spending most of the time interpreting softmask xobjects when drawing paths."

**Comment by JorjMcKie** [comment]
Hm, this is probably just an intermediate finding. I am sure they are still working on it.
So let's hang on ...

**Comment by naktinis** [comment]
> I have just submitted the bug report and will get back to you, when I get a reply there.

Just wanted to leave a link to the submitted bug here: 
https://bugs.ghostscript.com/show_bug.cgi?id=703060

It also seems it hasn't been addressed yet on MuPDF for the last 3.5 years and is marked as "UNCONFIRMED".
